From e4e8b16e66f592749984b2d287e0479e93ef35b3 Mon Sep 17 00:00:00 2001
From: Beleswar Padhi <b-padhi@ti.com>
Date: Tue, 3 Dec 2024 20:26:52 +0530
Subject: [PATCH 2/2] arm: dts: am57xx: Extend IPU Late Attach support to all
 AM57xx platforms

Late Attach feature of IPU remoteproc was added in only am574x-idk
platform. Extend the Late Attach support to all AM57xx platforms by
including the "dra7-ipu-common-early-boot.dtsi" file in the
corresponding DT files, as Early Boot support has been enabled in all
AM57xx platforms.

Fixes: 28ccf9733bd2 ("HACK: arm: dts: dra7/am57: Enable late attach support for IPU1 rproc")
Signed-off-by: Beleswar Padhi <b-padhi@ti.com>
---
 arch/arm/boot/dts/am571x-idk.dts                | 2 ++
 arch/arm/boot/dts/am572x-idk-common.dtsi        | 2 ++
 arch/arm/boot/dts/am574x-idk.dts                | 2 --
 arch/arm/boot/dts/am57xx-beagle-x15-common.dtsi | 2 ++
 4 files changed, 6 insertions(+), 2 deletions(-)

diff --git a/arch/arm/boot/dts/am571x-idk.dts b/arch/arm/boot/dts/am571x-idk.dts
index ce40c5d8d1425..4eeb2510f6d5c 100644
--- a/arch/arm/boot/dts/am571x-idk.dts
+++ b/arch/arm/boot/dts/am571x-idk.dts
@@ -284,3 +284,5 @@ &pruss2_eth {
 	ti,pruss-gp-mux-sel = <4>,	/* MII2, needed for PRUSS1_MII0 */
 			      <4>;	/* MII2, needed for PRUSS1_MII1 */
 };
+
+#include "dra7-ipu-common-early-boot.dtsi"
diff --git a/arch/arm/boot/dts/am572x-idk-common.dtsi b/arch/arm/boot/dts/am572x-idk-common.dtsi
index 1d66278c3a722..4005515f1ba99 100644
--- a/arch/arm/boot/dts/am572x-idk-common.dtsi
+++ b/arch/arm/boot/dts/am572x-idk-common.dtsi
@@ -201,3 +201,5 @@ &dsp2 {
 	status = "okay";
 	memory-region = <&dsp2_memory_region>;
 };
+
+#include "dra7-ipu-common-early-boot.dtsi"
diff --git a/arch/arm/boot/dts/am574x-idk.dts b/arch/arm/boot/dts/am574x-idk.dts
index 9eeed5f8056e8..47b9174d23530 100644
--- a/arch/arm/boot/dts/am574x-idk.dts
+++ b/arch/arm/boot/dts/am574x-idk.dts
@@ -47,5 +47,3 @@ &pruss1_mdio {
 &pruss2_mdio {
 	status = "disabled";
 };
-
-#include "dra7-ipu-common-early-boot.dtsi"
diff --git a/arch/arm/boot/dts/am57xx-beagle-x15-common.dtsi b/arch/arm/boot/dts/am57xx-beagle-x15-common.dtsi
index 994e69ab38d72..f08a1c54cd84d 100644
--- a/arch/arm/boot/dts/am57xx-beagle-x15-common.dtsi
+++ b/arch/arm/boot/dts/am57xx-beagle-x15-common.dtsi
@@ -645,3 +645,5 @@ &pruss1_mdio {
 &pruss2_mdio {
 	status = "disabled";
 };
+
+#include "dra7-ipu-common-early-boot.dtsi"
-- 
2.47.0

